File: Backend/AstroGame.Shared/Models/Technologies/StellarObjectDependentFinishedTechnology.cs
using AstroGame.Shared.Models.Players;
using System;

namespace AstroGame.Shared.Models.Technologies
{
    public class StellarObjectDependentFinishedTechnology : FinishedTechnology
    {
        public Guid ColonizedStellarObjectId { get; set; }

        public virtual ColonizedStellarObject ColonizedStellarObject { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/technologies/stellar_object_dependent_finished_technology.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/technologies/finished_technology.dart';
import 'package:flutter_guid/flutter_guid.dart';
import 'package:json_annotation/json_annotation.dart';

part 'stellar_object_dependent_finished_technology.g.dart';

@GuidConverter()
@JsonSerializable()
class StellarObjectDependentFinishedTechnology extends FinishedTechnology {
  @JsonKey()
  Guid colonizedStellarObjectId;

  StellarObjectDependentFinishedTechnology();

  factory StellarObjectDependentFinishedTechnology.fromJson(
          Map<String, dynamic> json) =>
      _$StellarObjectDependentFinishedTechnologyFromJson(json);
  Map<String, dynamic> toJson() =>
      _$StellarObjectDependentFinishedTechnologyToJson(this);
}
